public/calendar_reboot.js

```diff
@@ -820,12 +887,48 @@ var LocalCalendarStorage = function(params) {
         return key;
     };
     
+    self.onActivityChanged = function(func) {
+        // Will run checkit's onActivityChanged.
+        
+        //TODO extend this to support multiple callbacks
+        self.activityChangeFunctions.push(func);
+    };
+    
+
+    self.startWork = function() {
+        // Will increment the counter and possibly fire an event.
+        
+        self.activeCalls += 1;
+        console.log("store's startWork is running, activeCalls = " + self.activeCalls);
+        
+        // Will dispatch the event backgroundActivityChange 
+        self.activityChangeFunctions.forEach(function(func) {
+            func(self.activeCalls);
+        });
+        
+    };
+    
+    self.endWork = function() {
+        // Will decrement the counter and maybe fire an event.
+        
+        self.activeCalls -= 1;
```

Comment:
Can you add an assertion that `activeCalls >= 0` here? I have a feeling it will be tricky to make sure there is exactly one invocation of `endWork` for every `startWork`.